src/backends/fvm_gpu.hpp

```diff
@@ -51,6 +53,14 @@ __global__ void matrix_solve(matrix_solve_param_pack<T, I> params);
 template <typename T, typename I>
 __global__ void assemble_matrix(matrix_update_param_pack<T, I> params, T dt);
 
+template <typename T, typename I, typename Stack>
+__global__
+void test_thresholds(
+    float t, float t_prev, int size,
```

Comment:
using `t` and `t_<suffix>` for time values is pretty common across the code; I don't think we need to qualify it further. Using `t` for something not a time would be a problem though.